### PR TITLE
Feature/string field

### PR DESCRIPTION
**Pull request by abhishek-ram**
PR for Issue #6 - String Field Impl

### CODE REVIEW

File: src/protean/core/field/__init__.py [hunk 1]
```diff
@@ -110,3 +123,29 @@ def validate(self, value: Any):
         self._run_validators(value)
 
         self.value = value
+
+
+class String(Field):
```
Comment (subhashb) [inline src/protean/core/field/__init__.py]: Create a separate file for this class, please.

File: tests/core/test_field_types.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+""" Test cases for all available field type implementations"""
+
+import pytest  # pylint: disable=E0401
+from protean.core import field
+from protean.core.exceptions import ValidationError
+
+
+class TestStringField:
```
Comment (subhashb) [inline tests/core/test_field_types.py]: Move this into a `String` specific test case file.

File: src/protean/core/field/__init__.py (hunk 1 above)
Comment (abhishek-ram) [reply src/protean/core/field/__init__.py]: Okay this causes a bit of issues as we want uses to be able to import String from Field. In order to do this I will need to move the Field to a base.py and then add String to __all__ in __init__.py.. What do you think

Comment (subhashb) [reply src/protean/core/field/__init__.py]: Would this structure work?
`src -> core -> fields -> field.py -> Field class`
`src -> core -> fields -> string.py -> String class`

Then we can include in init.py approapriately to have:
`from protean.core.fields import Field, String`

**Pull request merged**